Some packages missing from `/embedded` in build

**Describe the bug**

In https://github.com/chromaui/chromatic-e2e/pull/282 we added a way to pack certain dependencies in `/embedded`. However it does not include some transitive dependencies. 

- `has-flag` is not in root of https://npmx.dev/package-code/@chromatic-com/playwright/v/0.13.0/embedded%2Fnode_modules.
- `chalk` > `supports-color` > `has-flag` at https://npmx.dev/package-code/@chromatic-com/playwright/v/0.13.0/embedded%2Fnode_modules%2Fchalk%2Fnode_modules%2Fsupports-color%2Fpackage.json#L45

This causes runtime issues if user's project doesn't contain `has-flag`.

This issue did not come up during testing, as projects under manual testing contained dependencies like `eslint` that depend on `chalk`, which brought `has-flag` in project's dependency tree.

The `resolvePackagePath` throws `Package subpath './package.json' is not defined by "exports"` for packages that have `exports` defined in `package.json`, but don't export `package.json` there:

https://github.com/chromaui/chromatic-e2e/blob/6dc1219f63929a6e83efaaa8ac1ba4024cac8395/scripts/prepare-embedded.ts#L21-L28


**To Reproduce**
Steps to reproduce the behavior:

1. Open https://stackblitz.com/~/edit/node-41f2ygly?file=package.json&view=editor
2. `npm install`
3. `npm run archive-storybook`

**Expected behavior**

No errors about `has-flag` are thrown.

**Screenshots**

```sh
~/projects/node-41f2ygly
❯ npm run archive-storybook

> storybook-playwright@0.0.0 archive-storybook
> CHROMATIC_ARCHIVE_LOCATION=./ archive-storybook


┌  storybook v10.2.13
│
▲  Failed to load preset:
│  "file:///home/projects/node-41f2ygly/node_modules/@chromatic-com/playwright/embedded/node_modules/@storybook/builder-webpack5/dist/presets/preview-preset.js"
│
■  Error: Cannot find module 'has-flag'
│  Require stack:
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/chalk/node_modules/supports-color/index.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/chalk/source/index.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/fork-ts-checker-webpack-plugin/lib/hooks/tap-error-to-log-message.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/fork-ts-checker-webpack-plugin/lib/plugin.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/fork-ts-checker-webpack-plugin/lib/index.js
│  at Module._resolveFilename
│  (https://node41f2ygly-eo2g.w-credentialless-st...
│  at defaultResolveImpl
│  (https://node41f2ygly-eo2g.w-credentialless-staticb...
│  at resolveForCJSWithHooks
│  (https://node41f2ygly-eo2g.w-credentialless-sta...
│  at Module._load
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz.c...
│  at TracingChannel.traceSync
│  (https://node41f2ygly-eo2g.w-credentialless-s...
│  at wrapModuleLoad
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz...
│  at Module.require
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz...
│  at require2
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz.com/b...
│  at _0x168fc6
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz.com/...
│  at Object.eval
│  (file://./node_modules/@chromatic-com/playwright/embedded/node_modules/chalk/node_modules/supports-color/index.js#cjs:4:17)
│
●  Starting...
│
▲  unable to find package.json for @chromatic-com/shared-e2e
│
▲  unable to find package.json for @rrweb/types
│
▲  unable to find package.json for @storybook/builder-webpack5
│
▲  unable to find package.json for @storybook/server
│
▲  unable to find package.json for @storybook/server-webpack5
│
▲  unable to find package.json for express
│
●  Using implicit CSS loaders
│
●  Using default Webpack5 setup
│
■  Failed to build the preview
│
■  Error: Cannot find module 'baseline-browser-mapping'
│  Require stack:
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/browserslist/index.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/webpack/lib/config/browserslistTargetHandler.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/webpack/lib/config/target.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/webpack/lib/config/defaults.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/webpack/lib/util/serialization.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/webpack/lib/util/makeSerializable.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/webpack/lib/WebpackError.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/webpack/lib/DefinePlugin.js
│  -
│  ./node_modules/@chromatic-com/playwright/embedded/node_modules/webpack/lib/index.js
│  at Module._resolveFilename
│  (https://node41f2ygly-eo2g.w-credentialless-st...
│  at defaultResolveImpl
│  (https://node41f2ygly-eo2g.w-credentialless-staticb...
│  at resolveForCJSWithHooks
│  (https://node41f2ygly-eo2g.w-credentialless-sta...
│  at Module._load
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz.c...
│  at TracingChannel.traceSync
│  (https://node41f2ygly-eo2g.w-credentialless-s...
│  at wrapModuleLoad
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz...
│  at Module.require
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz...
│  at require2
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz.com/b...
│  at _0x168fc6
│  (https://node41f2ygly-eo2g.w-credentialless-staticblitz.com/...
│  at Object.eval
│  (file://./node_modules/@chromatic-com/playwright/embedded/node_modules/browserslist/index.js#cjs:1:86)
│
▲  Broken build, fix the error above.
│  You may need to refresh the browser.
│
```

**Additional context**

-

--- a/.changeset/salty-sheep-boil.md
+++ b/.changeset/salty-sheep-boil.md
@@ -1,0 +1,7 @@
+---
+'@chromatic-com/cypress': patch
+'@chromatic-com/playwright': patch
+'@chromatic-com/shared-e2e': patch
+---
+
+Include missing packages in embedded directory

--- a/scripts/prepare-embedded.ts
+++ b/scripts/prepare-embedded.ts
@@ -22,7 +22,23 @@ function resolvePackagePath(req: NodeRequire, packageId: string): string | null 
   try {
     const p = req.resolve(`${packageId}/package.json`);
     return path.dirname(p);
-  } catch {
+  } catch (error: any) {
+    /*
+     * Attempt to parse package.json from error message:
+     * ```
+     * Package subpath './package.json' is not defined by "exports" in /x/chromatic-e2e/node_modules/supports-color/package.json
+     * ```
+     */
+    if (error.code === 'ERR_PACKAGE_PATH_NOT_EXPORTED') {
+      const parsed = error.message.split('is not defined by "exports" in ')[1];
+
+      if (fs.existsSync(parsed)) {
+        return path.dirname(parsed);
+      }
+    }
+
+    console.log(styleText('bgRed', `Failed to resolve package.json for "${packageId}".`));
+
     return null;
   }
 }

